### PR TITLE
Feature/permission control

### PR DESCRIPTION
**Pull request by Tubyhes**
@yulrizka can you QA?

### CODE REVIEW

File: sense platform/sensors/CSLocationProvider.m [hunk 1]
```diff
@@ -192,34 +193,40 @@ - (BOOL) isEnabled {
...
+// enabled the location provider
 - (void) setIsEnabled:(BOOL) enable {
 	[self enableLocationUpdates:enable];
 	isEnabled = enable;
 }
 
+// notification handler for switching on the locationProvider to make the location provider run
+- (void) setEnabledNotificationHandler: (NSNotification*) notification {
+    NSLog(@"[LocationProvider] setEnabledNotificaiton received!");
+    [self setIsEnabled: TRUE];
```
Comment (yulrizka) [inline sense platform/sensors/CSLocationProvider.m]: the idiom is YES

Comment (Tubyhes) [inline sense platform/sensors/CSLocationProvider.m]: :(

**Comment by yulrizka** [comment]
I think we should also provide method to check whether the request is accepted or not. 
This would make it easier for developer to hide/show the application permission. 

Of course they could also do it them self on the app layer. But I could imagine brightr, dogeprober goalie will have to implement this themselves.

File: sense platform/sensors/CSLocationProvider.m (hunk 1 above)
Comment (Tubyhes) [inline sense platform/sensors/CSLocationProvider.m]: is the convention to use NO and YES always instead of FALSE and TRUE? Or only in specific cases?

**Comment by Tubyhes** [comment]
I guess we could build in something like that, makes more sense now that we can let the app force permission request. However, what does the app do now when it doesnt get the required permissions? Does it just crash? Does it stop running?

File: sense platform/sensors/CSLocationProvider.m (hunk 1 above)
Comment (yulrizka) [inline sense platform/sensors/CSLocationProvider.m]: (BOOL) ->  ObjC. basically signed char with value YES/NO
bool intrinsict boolean with value true/false or TRUE/FALSE

**Comment by yulrizka** [comment]
tried it, it doesn't crash. but without the location provider enabled i think it will be killed by os.
to have that func it's also nice because the next time user brings up the app. it will still be reminded to enable the location permission. 

Although i also test that after rejecting the permission. and press allow again, it doesn't show the permission dialog.

**Comment by yulrizka** [comment]
Anyhow i check the functionality, everything seems to work. when permission is accepted. location provider are enabled

**Comment by Tubyhes** [comment]
@yulrizka could you QA again?

**Comment by jorishjanssen** [comment]
@Tubyhes I have added some comments. Some might be considered a bit nagging/annoying but it is good to be critical especially since it is your first iOS experience :)

**Comment by Tubyhes** [comment]
@jorishjanssen processed your comments. Can you re-QA?

**Pull request merged**